src/libsyntax/parse/parser/stmt.rs

```diff
@@ -473,4 +492,61 @@ impl<'a> Parser<'a> {
             "this was erroneously allowed and will become a hard error in a future release"
         }).emit();
     }
+
+    fn recover_stmt_local(
+        &mut self,
+        span: Span,
+        attrs: Vec<Attribute>,
+        msg: &str,
+    ) -> PResult<'a, Option<Stmt>> {
+        let local = self.parse_local(attrs.into())?;
+        self.struct_span_err(span, "invalid variable declaration")
+            .span_suggestion_short(
+                span,
+                msg,
+                "let".to_string(),
```

Comment:
```suggestion
                "let mut".to_string(),
```
The current code should transform `mut x = 0;` to `let x = 0;`.

Reply:
I think the appropriate thing to do here is to make the span for `span_suggestion_short` be `span.shrink_to_lo()` and the suggested code `"let ".to_string()` (notice the trailing space).